Public/Src/Tools/UnitTests/Analyzers/XLGToDBAnalyzerTests.cs

```diff
@@ -69,5 +74,62 @@ public void DBCreatedAndPopulated()
 
             XAssert.AreNotEqual(Directory.GetFiles(OutputDirPath.ToString(Context.PathTable), "*.sst").Length, 0);
         }
+
+        /// <summary>
+        /// This test makes sure that the number of events logged (per event type) match 
+        /// what is expected based on the build schedule that we manually populate and build.
+        /// </summary>
+        [Fact]
+        public void DBEventCountVerification()
+        {
+            Configuration.Logging.LogExecution = true;
+
+            var fileOne = FileArtifact.CreateOutputFile(Combine(TestDirPath, "foo.txt"));
+            var fileTwo = FileArtifact.CreateOutputFile(Combine(TestDirPath, "bar.txt"));
+            var dirOne = CreateOutputDirectoryArtifact(Combine(TestDirPath, "baz").ToString(Context.PathTable));
+            Directory.CreateDirectory(ArtifactToString(dirOne));
+
+            var pipA = CreateAndSchedulePipBuilder(new Operation[]
+            {
+                Operation.WriteFile(fileOne),
+            }).Process;
+
+            var pipB = CreateAndSchedulePipBuilder(new Operation[]
+            {
+                Operation.WriteFile(fileTwo),
+            }).Process;
+
+            var pipC = CreateAndSchedulePipBuilder(new Operation[]
+            {
+                Operation.ReadFile(fileOne),
+                Operation.WriteFile(CreateOutputFileArtifact())
+            }).Process;
+
+            var pipD = CreateAndSchedulePipBuilder(new Operation[]
+            {
+                Operation.EnumerateDir(dirOne),
+                Operation.WriteFile(CreateOutputFileArtifact())
+            }).Process;
+
+            var buildA = RunScheduler().AssertSuccess();
+
+            var analyzerRes = RunAnalyzer(buildA).AssertSuccess();
+
+            var dataStore = new XldbDataStore(storeDirectory: OutputDirPath.ToString(Context.PathTable));
+
+            XAssert.AreEqual(dataStore.GetFileArtifactContentDecidedEvents().Count(), 5);
+            XAssert.AreEqual(dataStore.GetWorkerListEvents().Count(), 0);
+            XAssert.AreEqual(dataStore.GetPipExecutionPerformanceEvents().Count(), 4);

```

Comment:
If you take my advice, here instead of `4` you could for sure say `LastGraph.RetrievePipReferencesOfType(PipType.Process).Count()`.  That would explicitly make it clear that we expect to get one PipExecutionPerformanceEvent per process pip.